### ISSUE TITLE
Incorrect return from Type.getInstanceFields

### ISSUE THREAD

**Issue by theJenix** [benmerckx/genes#77]
When generating Javascript code using genes, Type.getInstanceFields returns the wrong result:

Expected: `InstanceFields.hx:11: [ 'foo', 'bar' ]`

Actual: `InstanceFields.hx:11: [ 'new' ]`

InstanceFields.hx
```
class MyType {
    var foo = 1;
    var bar = 2;
    public function new() {}
}

class InstanceFields {

    static function main() {
        trace(Type.getInstanceFields(MyType));

    }
}
```

build.hxml
```
-lib genes:0.4.10

-D dts
-D genes.enum_discriminator=_name

-js build/instancefields.js
-D js-es=6
-D eval-stack
--main InstanceFields 
```

InstanceFields.js
```
export const MyType = Register.global("$hxClasses")["MyType"] = 
class MyType extends Register.inherits() {
	new() {
		this.bar = 2;
		this.foo = 1;
	}
	static get __name__() {
		return "MyType"
	}
	get __class__() {
		return MyType
	}
}


export const InstanceFields = Register.global("$hxClasses")["InstanceFields"] = 
class InstanceFields {
	static main() {
		console.log("InstanceFields.hx:11:",Type.getInstanceFields(MyType));
	}
	static get __name__() {
		return "InstanceFields"
	}
	get __class__() {
		return InstanceFields
	}
}
```

To run, build these files, add InstanceFields.main() to the bottom of InstanceFields.js (see related issue #78 ), and then run `node build/InstanceFields.js`.  I had to add a package.json to my root for es6 module support, also:
```
{
  "name": "instancefields",
  "version": "1.0.0",
  "description": "",
  "main": "instancefields.js",
  "type": "module",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
  }
}
```

**Comment by theJenix** [comment]
It looks like the issue might be genes is not assigning the fields to the MyType.prototype (which is what Type.getInstanceFields uses to look up available instance fields).  Here's the output from that same compilation with the default JS generator:
```
// Generated by Haxe 4.3.3
(function ($global) { "use strict";
class MyType {
	constructor() {
		this.bar = 2;
		this.foo = 1;
	}
}
Object.assign(MyType.prototype, {
	foo: null
	,bar: null
});
class InstanceFields {
	static main() {
		console.log("InstanceFields.hx:11:",Type.getInstanceFields(MyType));
	}
}
class Type {
	static getInstanceFields(c) {
		let result = [];
		while(c != null) {
			let _g = 0;
			let _g1 = Object.getOwnPropertyNames(c.prototype);
			while(_g < _g1.length) {
				let name = _g1[_g];
				++_g;
				switch(name) {
				case "__class__":case "__properties__":case "constructor":
					break;
				default:
					if(result.indexOf(name) == -1) {
						result.push(name);
					}
				}
			}
			c = c.__super__;
		}
		return result;
	}
}
class haxe_iterators_ArrayIterator {
	constructor(array) {
		this.current = 0;
		this.array = array;
	}
	hasNext() {
		return this.current < this.array.length;
	}
	next() {
		return this.array[this.current++];
	}
}
Object.assign(haxe_iterators_ArrayIterator.prototype, {
	array: null
	,current: null
});
{
}
InstanceFields.main();
})({});
```

This part specifically I think is what's missing to make this work:
```
Object.assign(MyType.prototype, {
	foo: null
	,bar: null
});
```

**Comment by benmerckx** [comment]
This should be fixed in bcc4063298e6fbac4c9e59f6d6bdd93876b9f829
I'll have a look at your other issue before releasing
